### ISSUE TITLE
`sinkhorn_divergence` does not handle `Grid` geometry

### ISSUE THREAD

**Issue by tlacombe** [ott-jax/ott#562]
**Describe the bug**

When trying to compute a Sinkhorn divergence with a `Grid` geometry, I get a (somewhat cryptic) error. It seems to be related to a call to `mean_cost_matrix` (which is indeed not defined on `Grid` geometry).

**To Reproduce**

```python
from jax import random, numpy as jnp
from ott.tools.sinkhorn_divergence import sinkhorn_divergence
from ott.geometry.grid import Grid

key = random.PRNGKey(0)
keys = random.split(key, 2)

gs = (5,)

a = random.uniform(keys[0], shape=gs)
a = a / jnp.sum(a)
b = random.uniform(keys[1], shape=gs)
b = b / jnp.sum(b)

sinkhorn_divergence(Grid(grid_size=gs), a=a, b=b)
```

returns

`TypeError: Shapes must be 1D sequences of concrete values of integer type, got (1.0,).`

**Expected behavior**

Either fixing the issue (I don't know how), or documenting that `sinkhorn_divergence` is not available for `Grid`, I guess. Right now, the documentation suggests that `sinkhorn_divergence` may take any type of geometry as input. 

**Complementary investigation:** 

Should it help, you can use the following MWE to raise the error: 

```python
# the following two lines are copy-pasted from the source code
geoms = Grid(grid_size=(5,)).prepare_divergences()   # this line of code does not return an error
geom_xy, geom_x, geom_y, *_ = geoms + (None,) * 3  # neither this one

geom_xy.mean_cost_matrix  # raises the error
```

**Comment by michalk8** [comment]
Hi @tlacombe , there was a bug, will be fixed in #563 .
Regarding the usage, this works:
```python
from jax import random, numpy as jnp
from ott.tools.sinkhorn_divergence import sinkhorn_divergence
from ott.geometry.grid import Grid

key = random.PRNGKey(0)
keys = random.split(key, 2)

gs = (5,)

a = random.uniform(keys[0], shape=gs)
a = a / jnp.sum(a)
b = random.uniform(keys[1], shape=gs)
b = b / jnp.sum(b)

sinkhorn_divergence(Grid, grid_size=gs, a=a, b=b)  # notice passing the class, not the instance
```

**Comment by tlacombe** [comment]
Hum, I still have the same error when I execute the cell you propose. I have `ott 0.4.7`, which I guess is the last version available (?) on pip (as I did run `pip install jax-ott --upgrade`). 

Not a very big deal (it was just to do some sanity check). If this is fixed in a next release, it's perfectly fine for me.
